torch/functional.py

```diff
@@ -102,12 +102,45 @@ def broadcast_shapes(*shapes):
     """
     # This wrapper exists to support variadic args.
     # TODO Movie this to C++ once the jit has better support for torch.Size.
-    with torch.no_grad():
-        scalar = torch.zeros((), device="cpu")
-        tensors = [scalar.expand(shape) for shape in shapes]
-        tensors = broadcast_tensors(*tensors)
-        return tensors[0].shape
-
+    max_len = 0
+    for shape in shapes:
+        if isinstance(shape, int):
+            if max_len < 1:
+                max_len = 1
+        elif isinstance(shape, tuple) or isinstance(shape, list):
+            s = len(shape)
+            if max_len < s:
+                max_len = s
+    result = [1] * max_len
+    max_num = 0
+    for shape in shapes:
+        if isinstance(shape, int):
+            if shape < 0:
+                raise RuntimeError(rf"Trying to create tensor with negative dimension {shape}: [{shape}]")
+            if max_num < shape:
```

Comment:
this logic doesn't work:
```
In [37]: broadcast_shapes(3,5)
Out[37]: torch.Size([5])
```
I suggest you wrap int in tuple:
```
if isinstance(shape, int):
    shape=(shape,)
```
and go directly to next branch

Reply:
Done :)